File: leetcode/js/1-bit-and-2-bit-characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function (bits) {
  const strBits = bits.join("");
  const replaced = strBits.replaceAll("11", "2").replaceAll("10", "2");

  return replaced.at(-1) === "0";
};

var isOneBitCharacter = function (bits) {
  let i = 0;

  while (i < bits.length - 1) {
    if (bits[i] === 0) {
      i += 1;
    } else {
      i += 2;
    }
  }

  return i === bits.length - 1;
};

File: leetcode/rust/1-bit-and-2-bit-characters.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut i = 0;

        while i < bits.len() - 1 {
            if bits[i] == 0 {
                i += 1;
            } else {
                i += 2;
            }
        }

        i == bits.len() - 1
    }
}
